tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/SymbolLookupTest.java

```diff
@@ -261,39 +261,40 @@ public void testSymbolLookupInQuery() {
         ModuleID moduleID = new BallerinaModuleID(pkg.packageID);
 
         Map<String, Symbol> symbolsForWhere = getSymbolsInFile(model, srcFile, 22, 21, moduleID);
-        assertList(symbolsForWhere, Arrays.asList("test", "arr1", "arr2", "i", "j"));
+        assertList(symbolsForWhere, Arrays.asList("test", "arr1", "arr2", "i", "j", "res1"));
 
         Map<String, Symbol> symbolsForSelect = getSymbolsInFile(model, srcFile, 28, 21, moduleID);
-        assertList(symbolsForSelect, Arrays.asList("test", "arr1", "arr2", "i", "j", "stringVal", "intVal", "res1"));
+        assertList(symbolsForSelect, Arrays.asList("intVal", "test", "res1", "stringVal", "res2", "i", "j", "arr2",
```

Comment:
Earlier with `SymbolEnv.cloneSymbolEnvForClosure()` it was filtering variables and narrowed types. Now that we got rid of that, it will cause the following for query expressions as well https://github.com/ballerina-platform/ballerina-lang/issues/30790.